main.go

```diff
@@ -57,14 +59,27 @@ func main() {
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
 	flag.BoolVar(&development, "development", true, "Configures the logger to use a development config (stacktraces on warnings, no sampling)")
+
 	opts := zap.Options{
 		Development: development,
 	}
 	opts.BindFlags(flag.CommandLine)
 	flag.Parse()
 
+	// set log level from environment variable if provided and no flag was set
+	if logLevel := os.Getenv("LOG_LEVEL"); logLevel != "" && opts.Level == nil {
+		level, valid := parseLogLevel(logLevel)
+		if !valid {
+			_, _ = fmt.Fprintf(os.Stderr, "Warning: Invalid LOG_LEVEL value '%s', using default level. Valid values: debug, info, warn, warning, error\n", logLevel)
+		} else {
+			opts.Level = level
+		}
+	}
+
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
+	ctrl.Log.Error(errors.New("test error"), "This is a test error message to verify logging setup")
```

Comment:
So, if it works it will output an error telling it is not an error. But if it is not?
1. Should we include an instruction that tells user must expect this output to prove the operator was setup correctly?
2. Can we use debug or info level here instead?
3. Could it be the final step of the function and output something like "all good"?